File: app/src/test/java/org/schulcloud/mobile/data/model/CommentTest.java
package org.schulcloud.mobile.data.model;

import org.junit.Before;
import org.junit.Test;

import static junit.framework.Assert.assertEquals;

public class CommentTest {
    private static final String ID = "ID";
    private static final String SUBMISSIONID = "1234";
    private static final String AUTHOR = "author";
    private static final String COMMENT = "comment";
    private static final String CREATEDAT = "invisible";

    private Comment comment;

    @Before
    public void setUp() throws Exception {
        comment = createNewComment();
    }

    @Test
    public void testGetProperties() {
        assertEquals(comment._id, ID);
        assertEquals(comment.author, AUTHOR);
        assertEquals(comment.comment, COMMENT);
        assertEquals(comment.submissionId, SUBMISSIONID);
        assertEquals(comment.createdAt, CREATEDAT);
    }

    public static Comment createNewComment() {
        Comment c = new Comment();
        c._id = ID;
        c.submissionId = SUBMISSIONID;
        c.author = AUTHOR;
        c.comment = COMMENT;
        c.createdAt = CREATEDAT;

        return c;
    }
}

File: app/src/test/java/org/schulcloud/mobile/models/homework/submission/CommentTest.kt
package org.schulcloud.mobile.models.homework.submission

import org.junit.Assert.*
import org.junit.Before
import org.junit.Test

class CommentTest {
    private companion object {
        const val ID = "id"
        const val SUBMISSIONID = "submissionId"
        const val AUTHOR = "author"
        const val COMMENT = "comment"
        const val CREATEDAT = "createdAt"
    }

    private val newComment: Comment
        get() = Comment().apply {
            id = ID
            submissionId = SUBMISSIONID
            author = AUTHOR
            comment = COMMENT
            createdAt = CREATEDAT
        }
    private lateinit var comment: Comment

    @Before
    fun setUp() {
        comment = newComment
    }

    @Test
    fun testGetProperties() {
        assertEquals(ID, comment.id)
        assertEquals(SUBMISSIONID, comment.submissionId)
        assertEquals(AUTHOR, comment.author)
        assertEquals(COMMENT, comment.comment)
        assertEquals(CREATEDAT, comment.createdAt)
    }
}
